packages/ui/src/components/token/AddTokenManualView.tsx

```diff
@@ -0,0 +1,295 @@
+                        </div>
+                    ) : (
+                        <>
+                            {/* ADDRESS */}
+                            <div className="flex flex-col flex-1 p-6 pb-0 space-y-1">
+                                <TextInput
+                                    appearance="outline"
+                                    label="Token Contract Address"
+                                    placeholder={
+                                        values.tokenAddress || "Address"
+                                    }
+                                    {...register("tokenAddress", {
+                                        onChange: (e) => {
+                                            onAddressChange(e.target.value)
+                                        },
+                                    })}
+                                    error={errors.tokenAddress?.message}
+                                    autoFocus={true}
+                                    maxLength={42}
+                                    defaultValue={values.tokenAddress}
+                                />
+                            </div>
+
+                            {/* SYMBOL */}
+                            <div className="flex flex-col flex-1 p-6 pb-0 space-y-1">
+                                <TextInput
+                                    appearance="outline"
+                                    label="Token Symbol"
+                                    placeholder={values.tokenSymbol || "ETH"}
+                                    defaultValue={values.tokenSymbol}
+                                    error={errors.tokenSymbol?.message}
+                                    {...register("tokenSymbol", {
+                                        onChange: (e) => {
+                                            onSymbolChange(e.target.value)
+                                        },
+                                    })}
+                                />
+                            </div>
+
+                            {/* DECIMALS */}
+                            <div className="flex flex-col flex-1 p-6 pb-0 space-y-1">
+                                <TextInput
+                                    appearance="outline"
+                                    label="Decimals of Precision"
+                                    placeholder={
+                                        values.tokenDecimals
+                                            ? values.tokenDecimals.toString()
+                                            : "18"
+                                    }
+                                    defaultValue={values.tokenDecimals || ""}
+                                    readOnly={values.tokenDecimals !== "0"}
+                                    error={errors.tokenDecimals?.message}
+                                    {...register("tokenDecimals", {
+                                        onChange: (e) => {
```

Comment:
Why do we have to this? The onchange should be automatically managed by the `react-hook-form`

Reply:
Changes done.